src/BuildScriptGenerator/DefaultEnvironment.cs

```diff
@@ -7,11 +7,28 @@
 using System.Collections;
 using System.Collections.Generic;
 using System.Linq;
+using Microsoft.Oryx.Common;
 
 namespace Microsoft.Oryx.BuildScriptGenerator
 {
     internal class DefaultEnvironment : IEnvironment
     {
+        public EnvironmentType Type
+        {
+            get
+            {
+                foreach (var entry in LoggingConstants.OperationNameSourceEnvVars)
```

Comment:
nit: consider caching it. How big can this IEnumerable be?

Reply:
not big at all (should always stay less than 10 in my opinion) - but cache won't hurt